### DISCUSSION TITLE
Run yarn build inside Dockerfile

### DISCUSSION THREAD

**Discussion by ThomasKientz** [swan-io/swan-partner-frontend#912] [Q&A] [answered]
Hi, 

In order to build and deploy the swan-partner-frontend (https://swan-io.github.io/swan-partner-frontend/build-deploy) we have to run `yarn build` before creating the docker image with `docker build`.

I am no expert in Docker, is it possible to modify the Dockerfile to include the `yarn build` so we don't have to run it before building the image ? Goal here is to use a service like render.com to deploy swan. Unfortunately [I can't run a build command when using docker with Render](https://docs.render.com/deploys#example-build-commands-for-each-runtime)

> You can’t set a build command for services that use Docker. Instead, Render either builds a custom image based on your Dockerfile or pulls a specified image from your container registry.

How would you achieve this ?

Thanks !

**Comment by bloodyowl** [comment]
generally we always at least some build steps prior to building the image, so that compilation is part of the CI. what's the issue with Render? not sure I get it

**Reply by ThomasKientz** [reply]
Render does not include a CI pipeline. It's like heroku, it connects to you git repo and deploy on every push. It is able to build a docker image but it can't run a command before building the image.

**Comment by ThomasKientz** [comment, marked as answer]
I managed to achieve this with the following Dockerfile 

```dockerfile
FROM node:22 AS pre-builder
WORKDIR /app
ADD ./ ./
RUN yarn install --pure-lockfile
RUN yarn build

FROM node:22 AS builder
WORKDIR /app
COPY --chown=node:node --from=pre-builder /app/server ./
RUN yarn install --pure-lockfile

FROM node:22
WORKDIR /app
COPY --chown=node:node --from=builder /app ./
CMD ["yarn", "start"]
EXPOSE 8080
```
